server/etcdserver/raft.go

```diff
@@ -646,5 +643,50 @@ func createConfigChangeEnts(lg *zap.Logger, ids []uint64, self uint64, term, ind
 		next++
 	}
 
+	promoteNodeFunc := func(id uint64) {
```

Comment:
This seems like a big change to me, I would expect some tests at least for this.

Reply:
I want to promote learner to leader when the other nodes all down. So I use --force-new-cluster when restart the learner. I hope the role of the node from learner to leader, and it runs ok. But when I restart the node I accpet a panic.
```
panic: removed all voters
```
But follower can run this commond ok, so I think it's a bug here.